README.md

```diff
@@ -23,6 +23,89 @@ Should you need to work on the core alone, use a traditional git workflow:
 $ git clone git@github.com:NYPL/Simplified-server-core.git core
 ```
 
+## Testing
+The github actions CI service runs the unit tests against Python 3.6, 3.7, 3.8 and 3.9 automatically using [tox](https://tox.readthedocs.io/en/latest/). 
+
+To run `pytest` unit tests locally, install `tox`.
+
+```
+pip install tox
+```
+
+Tox has an environment for each python version and an optional `-docker` factor that will automatically use docker to
+deploy service containers used for the tests. You can select the environment you would like to test with the tox `-e` 
+flag.
+
+### Environments
+
+| Environment | Python Version |
+| ----------- | -------------- |
+| py36        | Python 3.6     |
+| py37        | Python 3.7     | 
+| py38        | Python 3.8     | 
+| py39        | Python 3.9     |
+
+All of these environments are tested by default when running tox. To test one specific environment you can use the `-e`
+flag. 
+
+Test Python 3.8
+```
+tox -e py38
+```
+
+### Docker
+
+If you install `tox-docker` tox will take care of setting up all the service containers necessary to run the unit tests
```

Comment:
This means that `tox-docker` is optional, right? If it's not installed, will `tox` run without issues? I haven't tested this yet but there are docker configs in the tox.ini file and I'm not sure if it'll just ignore those configs.

Reply:
Yes. `tox-docker` is optional. Tox just ignores that config if you don't have docker installed.

The [`envlist` line in tox.ini](https://github.com/NYPL-Simplified/server_core/pull/1249/files#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449R2) does mean that tox will try to run with the docker configuration by default (if you don't pass any environment flags to it). We could change this to not default to running with docker if that is preferable.

I like `tox-docker` as a solution here, since it makes it fairly easy to run the checks the same way as they are run in CI if something is failing in GH actions, but not failing for you locally. It also makes it really easy to run the tests if you are new to the codebase.

Reply:
Yeah, I agree about using `tox-docker`. Maybe you can say that installing `tox-docker` is preferable when testing locally so that it parallels GH actions.